### PR TITLE
Add Mindesteinkommen to Wohngeld and calculate number of children below 24.

### LINKED ISSUE #592: Minimum income check Wohngeld eligibility
There is an income check for Wohngeld. I think this is not implemented yet.

### PR DESCRIPTION
**Pull request by MImmesberger**
Based on #772 

Closes #592.

Also, this PR adds a function that computes the number of children below 24.

### CODE REVIEW

**Comment by MImmesberger** [comment]
Regarding the Wohngeld tests, I let the Mindesteinkommen test pass whenever the test comes from an official calculator (the BMWSB calculator doesn't have a wealth and income check) and I adjusted the results whenever the test seems to be made up.

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py [hunk 1]
```diff
@@ -38,41 +38,62 @@ def arbeitsl_geld_2_eink_m(
...
+        arbeitsl_geld_2_nettoeink_mit_freibetrag_m
         + kind_unterh_erhalt_m
         + unterhaltsvors_m
         + kindergeld_zur_bedarfsdeckung_m
         + kindergeldübertrag_m
     )
 
 
-def _arbeitsl_geld_2_nettoeink_ohne_transfers_m(  # noqa: PLR0913
+def arbeitsl_geld_2_nettoeink_mit_freibetrag_m(
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py:49]:
```suggestion
def arbeitsl_geld_2_nettoeink_nach_abzug_freibetrag_m(
```
Sorry for the last-minute change. I think that we should be very verbose here.

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py [hunk 2]
```diff
@@ -38,41 +38,62 @@ def arbeitsl_geld_2_eink_m(
...
+
+    Returns
+    -------
+    Income after taxes, social insurance contributions, and other deductions.
+
+    """
+    return arbeitsl_geld_2_nettoeink_ohne_freibetrag_m - arbeitsl_geld_2_eink_anr_frei_m
+
+
+def arbeitsl_geld_2_nettoeink_ohne_freibetrag_m(
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py:73]:
```suggestion
def arbeitsl_geld_2_nettoeink_vor_abzug_freibetrag_m(
```

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py [hunk 3]
```diff
@@ -38,41 +38,62 @@ def arbeitsl_geld_2_eink_m(
...
 
     Note: Since 2023, Arbeitslosengeld 2 is referred to as Bürgergeld.
 
     Parameters
     ----------
     arbeitsl_geld_2_bruttoeink_m
         See :func:`arbeitsl_geld_2_bruttoeink_m`.
-    sozialv_beitr_arbeitnehmer_m
-        See :func:`sozialv_beitr_arbeitnehmer_m`.
     eink_st_y_sn
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py:89]:
```suggestion
    eink_st_m_sn
```
Or am I missing something again? :scream:

File: src/_gettsim/transfers/benefit_checks/benefit_checks.py [hunk 4]
```diff
@@ -26,16 +26,16 @@ def wohngeld_vorrang_bg(
...
-        See :func:`wohngeld_nach_vermög_check_m_bg`.
+    wohngeld_anspruchshöhe_m_bg
+        See :func:`wohngeld_anspruchshöhe_m_bg`.
 
     Returns
     -------
 
     """
     return (
         arbeitsl_geld_2_regelbedarf_m_bg
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/benefit_checks/benefit_checks.py:37]: I know it is a tiny thing, but given it is called `wohngeld_vorrang`, I have a strong opinion the check should go the other direction. 
```suggestion
        arbeitsl_geld_2_eink_m_bg + wohngeld_anspruchshöhe_m_bg >= arbeitsl_geld_2_regelbedarf_m_bg
```

Same for the two functions immediately below.

File: src/_gettsim/transfers/benefit_checks/benefit_checks.py [hunk 5]
```diff
@@ -26,16 +26,16 @@ def wohngeld_vorrang_bg(
...
-        <= arbeitsl_geld_2_eink_m_bg + wohngeld_nach_vermög_check_m_bg
+        <= arbeitsl_geld_2_eink_m_bg + wohngeld_anspruchshöhe_m_bg
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/benefit_checks/benefit_checks.py:38]:
```suggestion
```

File: src/_gettsim/transfers/kinderzuschl/kinderzuschl.py [hunk 6]
```diff
@@ -4,6 +4,13 @@
 
 from _gettsim.shared import policy_info
 
+aggregate_by_group_kinderzuschl = {
+    "_kinderzuschl_nach_vermög_check_m_wthh": {
+        "source_col": "_kinderzuschl_nach_vermög_check_m_for_aggregation",
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/kinderzuschl/kinderzuschl.py:9]: Shouldn't this be

```suggestion
        "source_col": "_kinderzuschl_nach_vermög_check_individual_level_m",
```

?

But why do we need to add anything at all? I'd think `_kinderzuschl_nach_vermög_check_m` should do, we do have custom converter functions after all?

File: src/_gettsim/transfers/wohngeld.py [hunk 7]
```diff
@@ -66,13 +66,151 @@ def wohngeld_m_wthh(
...
 
+@policy_info(params_key_for_rounding="wohngeld")
+def wohngeld_anspruchshöhe_m_wthh(
+    anz_personen_wthh: int,
+    wohngeld_eink_m_wthh: float,
+    wohngeld_miete_m_wthh: float,
+    wohngeld_anspruchsbedingungen_erfüllt_wthh: bool,
+    wohngeld_params: dict,
+) -> float:
+    """Housing benefit after wealth and income check.
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/wohngeld.py:84]:
```suggestion
    """Housing benefit.
```

File: src/_gettsim/transfers/wohngeld.py [hunk 8]
```diff
@@ -66,13 +66,151 @@ def wohngeld_m_wthh(
...
+
+@policy_info(params_key_for_rounding="wohngeld")
+def wohngeld_anspruchshöhe_m_bg(
+    anz_personen_bg: int,
+    wohngeld_eink_m_bg: float,
+    wohngeld_miete_m_bg: float,
+    wohngeld_anspruchsbedingungen_erfüllt_bg: bool,
+    wohngeld_params: dict,
+) -> float:
+    """Housing benefit after wealth and income check.
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/wohngeld.py:128]:
```suggestion
    """Housing benefit.
```

File: src/_gettsim/transfers/wohngeld.py [hunk 9]
```diff
@@ -691,21 +829,15 @@ def wohngeld_miete_ab_2009_m_hh(  # noqa: PLR0912 (see #516)
...
     vermögen_bedürft_wthh: float,
     anz_personen_wthh: int,
     wohngeld_params: dict,
-) -> float:
-    """Preliminary housing benefit after wealth check.
-
-    This target is used for the actual Wohngeld calculation of the Bedarfsgemeinschaften
-    that passed the priority check against ALG2 and Kinderzuschlag.
+) -> bool:
+    """Wealth is below the threshold for housing benefit.
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/wohngeld.py:837]:
```suggestion
    """Wealth is below the eligibility threshold for housing benefits.
```

File: src/_gettsim/transfers/wohngeld.py [hunk 10]
```diff
@@ -718,29 +850,22 @@ def _wohngeld_nach_vermög_check_m_wthh(
...
     vermögen_bedürft_bg: float,
     anz_personen_bg: int,
     wohngeld_params: dict,
-) -> float:
-    """Preliminary housing benefit after wealth check.
-
-    This target is used for the priority check calculation against ALG2 and
-    Kinderzuschlag.
+) -> bool:
+    """Wealth is below the threshold for housing benefit.
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/wohngeld.py:865]:
```suggestion
    """Wealth is below the eligibility threshold for housing benefits.
```

File: src/_gettsim/transfers/wohngeld.py [hunk 11]
```diff
@@ -753,83 +878,115 @@ def wohngeld_nach_vermög_check_m_bg(
...
-    wohngeld_miete_m_wthh: float,
-    wohngeld_params: dict,
-) -> float:
-    """Housing benefit on wohngeldrechtlicher Teilhaushalt level without wealth or
-    priority checks.
+def wohngeld_mindesteinkommen_erreicht_wthh(
+    arbeitsl_geld_2_regelbedarf_m_wthh: float,
+    wohngeld_einkommen_für_mindesteinkommen_check_m_wthh: float,
+) -> bool:
+    """Minimum income requirement for Wohngeld has been met.
```
Comment (hmgaudecker) [inline src/_gettsim/transfers/wohngeld.py:892]:
```suggestion
    """Minimum income requirement for housing benefits is met.
```

File: src/_gettsim_tests/test_data/arbeitsl_geld_2/2005/hh_id_8.yaml [hunk 12]
```diff
@@ -66,9 +66,9 @@ inputs:
       - 0.0
     jahr:
       - '2005'
-    wohngeld_vor_vermög_check_m_wthh:
+    wohngeld_anspruchshöhe_m_bg:
       - 0.0
-    wohngeld_vor_vermög_check_m_bg:
+    wohngeld_basisbetrag_m_bg:
```
Comment (hmgaudecker) [inline src/_gettsim_tests/test_data/arbeitsl_geld_2/2005/hh_id_8.yaml:71]: I think this is gone and can be deleted in all tests?

**Review by hmgaudecker — APPROVED** [review]
Excellent, thanks a lot! 

Just a couple of renaming suggestions and one substantive question, nothing which would stop anything.

File: src/_gettsim_tests/test_data/arbeitsl_geld_2/2005/hh_id_8.yaml (hunk 12 above)
Comment (MImmesberger) [reply src/_gettsim_tests/test_data/arbeitsl_geld_2/2005/hh_id_8.yaml:71]: Oh yes, no idea how that survived.

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py (hunk 3 above)
Comment (MImmesberger) [reply src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py:89]: Oh, yes! I was sure this doesn't work for some reason

**Comment by hmgaudecker** [comment]
Excellent, thanks! Feel free to merge!

**Pull request merged**